QUIRKS.md

```diff
@@ -0,0 +1,67 @@
+
+## Handling out-of-order frames
+
+### Problem
+
+LPMS transcoder used to fail when segments or frames come in messed order. This issue is caused when one segment failed to get uploaded to Transcoder due to poor network and gets delivered later in its retry attempts or when some frames get dropped due to poor network. The FPS filter expects the timestamps to increase monotonically and uniformly. If this requirement is not met, transcoding pipeline fails.
+
+### Solution
+
+```
+                                               FILTERGRAPH
+                         +------------------------------------------------------+
+                         |                                                      |
+                         |      +------------+              +------------+      |
+                         |      |            |              |            |      |
+                         |      |            |              |            |      |
+  +-----------+          |      |  SCALE     |              |  FPS       |      |         +-----------+
+  |           |          |      |   filter   |              |   filter   |      |         |           |
+  | decoded F +---------------->+            +------+------>+            +--------------->+ encoded F |
+  |           |          |      |            |      ^       |            |      |         |           |
+  +-----------+          |      |            |      |       |            |      |         +-----------+
+                         |      |            |      |       |            |      |
+     pts_in              |      |            |      |       |            |      |             pts_out
+                         |      +------------+      |       +------------+      |    (2)
+(non-monotonic &         |                          |                           |    (guess using pts_in & fps
+ unreliable jumps)       +------------------------------------------------------+       to maintain same order)
+                                                    |
+                                                    |
+                                                    |
+                                                    +
+                                           (1) dummy monotonic
+                                                   pts
+```
+
+FPS filter expects monotonic increase in input frame's PTS. We cannot rely on the input to be monotonic thus:
+i. Set a dummy PTS before the frame is sent into filtergraph, that we manually increase monotonically.
+ii. OR use SETPTS filter in the filtergraph before FPS filter, which would do the same thing.

```

Comment:
We should remove this from the documentation, as we did not take the approach of SETPTS filter due to reasons mentioned here https://github.com/livepeer/lpms/pull/201#issuecomment-664853222